sale_stock_picking_blocking/models/sale_order.py

```diff
@@ -18,13 +18,15 @@ class SaleOrder(models.Model):
         states={"draft": [("readonly", False)], "sent": [("readonly", False)]},
     )
 
-    @api.constrains("delivery_block_id")
-    def _check_not_auto_done(self):
-        auto_done = self.user_has_groups("sale.group_auto_done_setting")
-        if auto_done and any(so.delivery_block_id for so in self):
-            raise ValidationError(
-                _('You cannot block a sale order with "auto_done_setting" ' "active.")
-            )
+    def write(self, vals):
+        if self.delivery_block_id:
```

Comment:
`self` is a recordset , you need to put `self.delivery_block_id` inside a loop, such as `for record in self:`

Reply:
anw, from the card's requirement, you can remove the constraint `def _check_not_auto_done(self):` totally. No need to move it into `def write()`